futures-core/src/future/either.rs

```diff
@@ -1,24 +1,24 @@
-use {task, Future, Stream, Poll};
+use {task, Stream, Poll};
 
 use core::mem::PinMut;
 
 use either::Either;
 
-impl<A, B> Future for Either<A, B>
-    where A: Future,
-          B: Future<Output = A::Output>
-{
-    type Output = A::Output;
+// impl<A, B> Future for Either<A, B>
+//     where A: Future,
+//           B: Future<Output = A::Output>
+// {
+//     type Output = A::Output;
 
-    fn poll(self: PinMut<Self>, cx: &mut task::Context) -> Poll<A::Output> {
-        unsafe {
-            match PinMut::get_mut(self) {
-                Either::Left(a) => PinMut::new_unchecked(a).poll(cx),
-                Either::Right(b) => PinMut::new_unchecked(b).poll(cx),
-            }
-        }
-    }
-}
+//     fn poll(self: PinMut<Self>, cx: &mut task::Context) -> Poll<A::Output> {
+//         unsafe {
+//             match PinMut::get_mut(self) {
+//                 Either::Left(a) => PinMut::new_unchecked(a).poll(cx),
+//                 Either::Right(b) => PinMut::new_unchecked(b).poll(cx),
+//             }
+//         }
+//     }
+// }
```

Comment:
Why was this removed?

Reply:
since either is defined in https://github.com/bluss/either-rs and Future in std it can't exist here because of orphan rules - i might write a pr moving this to the either crate as an unstable feature

Reply:
I'd gently suggest removing the commented code, if it is, in fact, going away